packages/rule-utils/src/icon.ts

```diff
@@ -0,0 +1,110 @@
+import { type IconifyLoaderOptions, type UniversalIconLoader, encodeSvgForCss } from '@iconify/utils'
+import { type Preset, type UnoGenerator, toArray } from '@unocss/core'
+import type { IconsOptions } from '@unocss/preset-icons'
+
+const COLLECTION_NAME_PARTS_MAX = 3
+
+// eslint-disable-next-line regexp/no-super-linear-backtracking
+export const iconFnRE = /icon\(\s*(['"])?(.*?)\1?\s*\)/g
+
+export function hasIconFn(str: string) {
+  return str.includes('icon(') && str.includes(')')
+}
+
+export async function transformIconString(uno: UnoGenerator, icon: string, color?: string) {
+  const presetIcons = uno.userConfig.presets?.flat()?.findLast(i => i.name === '@unocss/preset-icons')
+
+  if (!presetIcons) {
+    console.warn('@unocss/preset-icons not found')
+    return
+  }
+
+  const {
+    scale = 1,
+    prefix = 'i-',
+    collections: customCollections,
+    extraProperties = {},
+    customizations = {},
+    autoInstall = false,
+    collectionsNodeResolvePath,
+    unit,
+  } = (presetIcons as Preset).options as IconsOptions
+
+  const loaderOptions: IconifyLoaderOptions = {
+    addXmlNs: true,
+    scale,
+    customCollections,
+    autoInstall,
+    cwd: collectionsNodeResolvePath,
+    // avoid warn from @iconify/loader: we'll warn below if not found
+    warn: undefined,
+    customizations: {
+      ...customizations,
+      additionalProps: { ...extraProperties },
+      trimCustomSvg: true,
+      async iconCustomizer(collection, icon, props) {
+        await customizations.iconCustomizer?.(collection, icon, props)
+        if (unit) {
+          if (!props.width)
+            props.width = `${scale}${unit}`
+          if (!props.height)
+            props.height = `${scale}${unit}`
+        }
+      },
+    },
+  }
+
+  const loader = await createNodeLoader()
+
+  for (const p of toArray(prefix)) {
+    if (icon.startsWith(p)) {
+      icon = icon.slice(p.length)
+      const parsed = await parseIcon(icon, loader, loaderOptions)
```

Comment:
Maybe it would be better to change it to parallel

Reply:
What I am considering is:
- In reality, the number of icon prefixes is generally not too large
- Parallel processing will result in redundant loaders for parsing. For my purpose, only one of them needs to have a value is enough.

Reply:
This will cause icon loading blockage, which may cause icon rendering to be too slow. In addition, this loader should be cached. In the case of cache, concurrency can have a faster advantage.